src/compiler/checker.ts

```diff
@@ -2750,7 +2750,15 @@ namespace ts {
             }
 
             function buildTypeParameterDisplay(tp: TypeParameter, writer: SymbolWriter, enclosingDeclaration?: Node, flags?: TypeFormatFlags, symbolStack?: Symbol[]) {
-                appendSymbolNameOnly(tp.symbol, writer);
+                const declaration = tp.symbol && tp.symbol.declarations && tp.symbol.declarations[0] as TypeParameterDeclaration;

```

Comment:
why do you need these? I would just not create a declaration, and use the `symbol.name`, same for parameters as well..